core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -159,11 +160,94 @@ describe('Contributor dashboard Admin page', () => {
     spyOn(
       contributorDashboardAdminStatsBackendApiService, 'fetchCommunityStats')
       .and.returnValue(Promise.resolve({
-        translation_reviewers_count: 1,
+        translation_reviewers_count: {
+          en: 1,
+          ar: 1
+        },
         question_reviewers_count: 1
-      } as CommunityContributionStatsDict));
+      } as CommunityContributionStatsBackendDict));
 
+    spyOn(
+      contributorDashboardAdminStatsBackendApiService,
+      'fetchAssignedLanguageIds')
+      .and.returnValue(Promise.resolve(['en', 'ar']));
+
+    spyOn(
+      contributorDashboardAdminStatsBackendApiService,
+      'fetchTopicChoices')
+      .and.returnValue(Promise.resolve([
+        { id: '1', topic: 'Science' },
+        { id: '2', topic: 'Technology' },
+      ]));
+    fixture.detectChanges();
+
+    component.ngOnInit();
+  }));
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('should initialize variables onInit', fakeAsync(() => {
+    spyOn(userService, 'getUserInfoAsync').and.returnValue(
+      Promise.resolve(fullAccessUserInfo));
+    component.ngOnInit();
+    tick();
+    fixture.detectChanges();
+    expect(component.selectedLanguage.language).toEqual('English');
+    expect(component.selectedLanguage.id).toEqual('en');
```

Comment:
```suggestion
    spyOn(userService, 'getUserInfoAsync').and.returnValue(
      Promise.resolve(fullAccessUserInfo));
      
    component.ngOnInit();
    tick();
    fixture.detectChanges();
   
    expect(component.selectedLanguage.language).toEqual('English');
    expect(component.selectedLanguage.id).toEqual('en');
```
Make sure to add new lines to differentiate the test phases (setup, action, assertion). This makes tests easy to scan and understand quickly. Ditto elsewhere.

Reply:
Done. Fixed everywhere. PTAL. Thanks.